src/test/FuzzerImpl.cpp

```diff
@@ -48,6 +48,72 @@ seedRandomness()
     gRandomEngine.seed(1);
 }
 
+// FuzzTransactionFrame is a specialized TransactionFrame that includes
+// useful methods for fuzzing such as an attemptApplication method for reseting
+// ledger state and deterministically attempting application of transactions.
+class FuzzTransactionFrame : public TransactionFrame
+{
+  public:
+    FuzzTransactionFrame(Hash const& networkID,
+                         TransactionEnvelope const& envelope)
+        : TransactionFrame(networkID, envelope){};
+
+    void
+    attemptApplication(Application& app, AbstractLedgerTxn& ltx)
+    {
+        // seed randomness
+        seedRandomness();
+
+        // before application, clear cache so that they don't persist
+        // between executions and confuse the fuzzer, causing it to
+        // think some Tx with [op0...opN] can result in multiple
+        // different execution paths.
+        app.getLedgerTxnRoot().resetForFuzzer();
+        app.getDatabase().clearPreparedStatementCache();
+
+        // reset results of operations
```

Comment:
This entire block looks nearly duplicate to `TransactionFrame::resetResults`. Can you refactor to share code?